{
  "text": "2021-07-26T10:21:03.182659+0600 ERROR An error has been caught in function 'dispatch_request', process 'MainProcess' (24094), thread 'Thread-2' (6168227840):\nTraceback (most recent call last):\n\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py\", line 912, in _bootstrap\n    self._bootstrap_inner()\n    \u2502    \u2514 <function Thread._bootstrap_inner at 0x102b659d0>\n    \u2514 <Thread(Thread-2, started daemon 6168227840)>\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py\", line 954, in _bootstrap_inner\n    self.run()\n    \u2502    \u2514 <function Thread.run at 0x102b65700>\n    \u2514 <Thread(Thread-2, started daemon 6168227840)>\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py\", line 892, in run\n    self._target(*self._args, **self._kwargs)\n    \u2502    \u2502        \u2502    \u2502        \u2502    \u2514 {}\n    \u2502    \u2502        \u2502    \u2502        \u2514 <Thread(Thread-2, started daemon 6168227840)>\n    \u2502    \u2502        \u2502    \u2514 (<socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 8080), raddr=('...\n    \u2502    \u2502        \u2514 <Thread(Thread-2, started daemon 6168227840)>\n    \u2502    \u2514 <bound method ThreadingMixIn.process_request_thread of <werkzeug.serving.ThreadedWSGIServer object at 0x1050367f0>>\n    \u2514 <Thread(Thread-2, started daemon 6168227840)>\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py\", line 650, in process_request_thread\n    self.finish_request(request, client_address)\n    \u2502    \u2502              \u2502        \u2514 ('127.0.0.1', 62926)\n    \u2502    \u2502              \u2514 <socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 8080), raddr=('1...\n    \u2502    \u2514 <function BaseServer.finish_request at 0x10352fd30>\n    \u2514 <werkzeug.serving.ThreadedWSGIServer object at 0x1050367f0>\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py\", line 360, in finish_request\n    self.RequestHandlerClass(request, client_address, self)\n    \u2502    \u2502                   \u2502        \u2502               \u2514 <werkzeug.serving.ThreadedWSGIServer object at 0x1050367f0>\n    \u2502    \u2502                   \u2502        \u2514 ('127.0.0.1', 62926)\n    \u2502    \u2502                   \u2514 <socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('127.0.0.1', 8080), raddr=('1...\n    \u2502    \u2514 <class 'werkzeug.serving.WSGIRequestHandler'>\n    \u2514 <werkzeug.serving.ThreadedWSGIServer object at 0x1050367f0>\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py\", line 720, in __init__\n    self.handle()\n    \u2502    \u2514 <function WSGIRequestHandler.handle at 0x103735940>\n    \u2514 <werkzeug.serving.WSGIRequestHandler object at 0x105036ca0>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/werkzeug/serving.py\", line 345, in handle\n    BaseHTTPRequestHandler.handle(self)\n    \u2502                      \u2502      \u2514 <werkzeug.serving.WSGIRequestHandler object at 0x105036ca0>\n    \u2502                      \u2514 <function BaseHTTPRequestHandler.handle at 0x1036c0c10>\n    \u2514 <class 'http.server.BaseHTTPRequestHandler'>\n  File \"/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/server.py\", line 427, in handle\n    self.handle_one_request()\n    \u2502    \u2514 <function WSGIRequestHandler.handle_one_request at 0x103735af0>\n    \u2514 <werkzeug.serving.WSGIRequestHandler object at 0x105036ca0>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/werkzeug/serving.py\", line 379, in handle_one_request\n    return self.run_wsgi()\n           \u2502    \u2514 <function WSGIRequestHandler.run_wsgi at 0x1037358b0>\n           \u2514 <werkzeug.serving.WSGIRequestHandler object at 0x105036ca0>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/werkzeug/serving.py\", line 323, in run_wsgi\n    execute(self.server.app)\n    \u2502       \u2502    \u2502      \u2514 <werkzeug.debug.DebuggedApplication object at 0x105002490>\n    \u2502       \u2502    \u2514 <werkzeug.serving.ThreadedWSGIServer object at 0x1050367f0>\n    \u2502       \u2514 <werkzeug.serving.WSGIRequestHandler object at 0x105036ca0>\n    \u2514 <function WSGIRequestHandler.run_wsgi.<locals>.execute at 0x10503fd30>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/werkzeug/serving.py\", line 314, in execute\n    for data in application_iter:\n                \u2514 <generator object DebuggedApplication.debug_application at 0x10503e040>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/werkzeug/debug/__init__.py\", line 304, in debug_application\n    app_iter = self.app(environ, start_response)\n               \u2502    \u2502   \u2502        \u2514 <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x10503fca0>\n               \u2502    \u2502   \u2514 {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=6>, 'wsgi.errors': <_io.TextIOWrap...\n               \u2502    \u2514 <Flask 'app'>\n               \u2514 <werkzeug.debug.DebuggedApplication object at 0x105002490>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/flask/app.py\", line 2464, in __call__\n    return self.wsgi_app(environ, start_response)\n           \u2502    \u2502        \u2502        \u2514 <function WSGIRequestHandler.run_wsgi.<locals>.start_response at 0x10503fca0>\n           \u2502    \u2502        \u2514 {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': <_io.BufferedReader name=6>, 'wsgi.errors': <_io.TextIOWrap...\n           \u2502    \u2514 <function Flask.wsgi_app at 0x1039b2ca0>\n           \u2514 <Flask 'app'>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/flask/app.py\", line 2447, in wsgi_app\n    response = self.full_dispatch_request()\n               \u2502    \u2514 <function Flask.full_dispatch_request at 0x1039b23a0>\n               \u2514 <Flask 'app'>\n  File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/flask/app.py\", line 1950, in full_dispatch_request\n    rv = self.dispatch_request()\n         \u2502    \u2514 <function Flask.dispatch_request at 0x1039b2310>\n         \u2514 <Flask 'app'>\n> File \"/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/flask/app.py\", line 1936, in dispatch_request\n    return self.view_functions[rule.endpoint](**req.view_args)\n           \u2502    \u2502              \u2502    \u2502           \u2502   \u2514 {}\n           \u2502    \u2502              \u2502    \u2502           \u2514 <Request 'http://127.0.0.1:8080/contact_us' [POST]>\n           \u2502    \u2502              \u2502    \u2514 'public.contact_us'\n           \u2502    \u2502              \u2514 <Rule '/contact_us' (GET, HEAD, OPTIONS, POST) -> public.contact_us>\n           \u2502    \u2514 {'static': <bound method _PackageBoundObject.send_static_file of <Flask 'app'>>, 'user.change_password': <function UserManage...\n           \u2514 <Flask 'app'>\n\n  File \"/Users/vadimganica/PycharmProjects/Diplom/app/public_bp/routes/core.py\", line 155, in contact_us\n    a / b\n    \u2502   \u2514 0\n    \u2514 3\n\nZeroDivisionError: division by zero\n",
  "record": {
    "elapsed": {
      "repr": "0:00:08.448356",
      "seconds": 8.448356
    },
    "exception": {
      "type": "ZeroDivisionError",
      "value": "division by zero",
      "traceback": true
    },
    "extra": {},
    "file": {
      "name": "app.py",
      "path": "/Users/vadimganica/PycharmProjects/Diplom/venv/lib/python3.9/site-packages/flask/app.py"
    },
    "function": "dispatch_request",
    "level": {
      "icon": "\u274c",
      "name": "ERROR",
      "no": 40
    },
    "line": 1936,
    "message": "An error has been caught in function 'dispatch_request', process 'MainProcess' (24094), thread 'Thread-2' (6168227840):",
    "module": "app",
    "name": "flask.app",
    "process": {
      "id": 24094,
      "name": "MainProcess"
    },
    "thread": {
      "id": 6168227840,
      "name": "Thread-2"
    },
    "time": {
      "repr": "2021-07-26 10:21:03.182659+06:00",
      "timestamp": 1627273263.182659
    }
  }
}
